app/src/main/java/com/ddoczi/tasky/agenda/presentation/home/HomeViewModel.kt

```diff
@@ -64,17 +75,32 @@ class HomeViewModel @Inject constructor(
             is HomeEvent.OnAgendaItemDismiss -> {
                 _state.update { it.copy(showAgendaOptions = false) }
             }
-            is HomeEvent.OnRefreshAgenda -> { Unit }
+            is HomeEvent.OnRefreshAgenda -> {
+                getAgendaForSelectedDate()
+            }
             is HomeEvent.OnDeleteItem -> {
                 viewModelScope.launch {
                     when (event.agendaItem) {
                         is AgendaItem.Event -> { eventRepository.deleteEventById(event.agendaItem.eventId) }
                         is AgendaItem.Task -> { taskRepository.deleteTaskById(event.agendaItem.taskId) }
                         is AgendaItem.Reminder -> { reminderRepository.deleteReminderById(event.agendaItem.reminderId) }
                     }
+                    getAgendaForSelectedDate()
                 }
             }
             else -> { Unit }
         }
     }
+    private fun getAgendaForSelectedDate() {
+        viewModelScope.launch {
+            val date = state.value.currentDate.plusDays(state.value.selectedDay.toLong())
+            val fromDate = date.atStartOfDay().atZone(ZoneId.systemDefault()).toInstant().toEpochMilli()
+            val toDate = date.atStartOfDay().plusDays(1).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli()
+            val tasks = taskRepository.getTasksByDate(fromDate, toDate)
+            val events = eventRepository.getEventsByDate(fromDate, toDate)
+            val reminders = reminderRepository.getRemindersByDate(fromDate, toDate)
```

Comment:
Are these suspending calls? Then it would be faster to make use of parallel coroutines here, since right now they're executing sequentially, one after another. You can use `async` blocks to do so